core/src/docs/rfcs/3898_concurrent_writer.md

```diff
@@ -0,0 +1,66 @@
+- Proposal Name: `concurrent_writer`
+- Start Date: 2024-01-02
+- RFC PR: [apache/incubator-opendal#3898](https://github.com/apache/incubator-opendal/pull/3898)
+- Tracking Issue: [apache/incubator-opendal#3899](https://github.com/apache/incubator-opendal/issues/3899)
+
+# Summary
+
+Enhance the `Writer` by adding concurrent write capabilities.
+
+# Motivation
+
+Currently, When invoking `writer` or `writer_with` on an `Operator` that utilizes Object storage as its backend, such as S3, the `Operator` will return a `Writer` that supports multipart uploading. 
```

Comment:
This sentence gives the impression that:

- Only object storage provides a writer.
- Object storage supports only multipart uploads.
- The term 'Writer' refers to multipart uploading.

The statement is false.

When users call `op.writer()`, we return a `Writer`; the implementation details are internal.

Certain services offer `multi_write` functionality, as detailed in [Capability::write_can_multi](https://opendal.apache.org/docs/rust/opendal/struct.Capability.html#structfield.write_can_multi), allowing users to execute multiple `write` operations for uploading large files.

Within services that support `multi_write`, such as S3, GCS, and AzBlob, concurrent writing is permitted.